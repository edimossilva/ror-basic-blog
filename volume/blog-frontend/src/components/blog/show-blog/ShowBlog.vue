<template lang="">
  <div>
    <h3>Show Blog</h3>

    <div>
      <h3> Blog Name: {{blog.name}} </h3>
    </div>
    
    <table>
      <thead>
        <tr>
          <th>id</th>
          <th>title</th>
          <th>show post</th>
          <th>delete</th>
        </tr>
      </thead>
      <tbody>
      <tr v-for="post in blog.posts" v-bind:key="post.id">
        <td>{{post.id}}</td>
        <td>{{post.title}}</td>
        <td><button @click="redirectToShowPost(post.id)">Show post</button></td>
        <td><button @click="deletePost(post.id)">Delete post</button></td>
      </tr>
    </tbody>
    </table>
    <button @click="redirectTocreatePost()">Create Post</button>

  </div>
</template>

<script>
import blogsApi from "../../../services/blogs_api";
import postsApi from "../../../services/posts_api";

export default {
  props: {
    id: {
      required: true
    }
  },

  data() {
    return {
      blog: {}
    };
  },

  methods: {
    deletePost(postId) {
      const { onDeletePost, showErrorMessage } = this;
      const params = {
        blogId: this.id,
        id: postId
      };

      postsApi
        .deletePost({ params, token: this.$store.getters.authToken })
        .then(onDeletePost)
        .catch(showErrorMessage);
    },
    onDeletePost() {
      console.log("deleted post");
    },
    getBlog() {
      const params = {
        id: this.id
      };
      blogsApi
        .getBlog({
          params,
          token: this.$store.getters.authToken
        })
        .then(this.updateData)
        .catch(this.showErrorMessage);
    },

    updateData({ data: { data } }) {
      this.blog = JSON.parse(data);
    },

    showErrorMessage({
      response: {
        data: { errors }
      }
    }) {
      debugger; // eslint-disable-line
      console.log(errors);
    },

    redirectToShowPost(postId) {
      this.$router.push({
        name: "showPost",
        params: { id: postId, blogId: this.blog.id }
      });
    },

    redirectTocreatePost() {
      this.$router.push({
        name: "createPost",
        params: { blogId: this.blog.id }
      });
    }
  },

  mounted: function() {
    this.getBlog();
  }
};
</script>

<style scoped></style>