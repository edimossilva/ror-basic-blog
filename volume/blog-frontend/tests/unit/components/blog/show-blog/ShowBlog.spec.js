import { shallowMount } from '@vue/test-utils'
import ShowBlog from '@/components/blog/show-blog/ShowBlog'

describe('ShowBlog.vue', () => {
  it('is a valid component', () => {
    const mockBlog = {
      name: "mockName",
      title: "mockTitle",
      private: "mockPrivate",

      posts: [
        { id: 1, owner: "mockOwner", title: "mockTitle1", is_private: true },
        { id: 2, owner: "mockOwner", title: "mockTitle2", is_private: false },
      ]
    }
    const wrapper = shallowMount(ShowBlog, {
      methods: { getBlog: jest.fn() },
      propsData: { id: 1 },
      data() { return { blog: mockBlog } }
    });

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
