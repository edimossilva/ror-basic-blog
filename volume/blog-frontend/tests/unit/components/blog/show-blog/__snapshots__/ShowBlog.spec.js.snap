// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowBlog.vue is a valid component 1`] = `
<div>
  <div>
    <h3> Blog Name: mockName </h3>
  </div>
  <table>
    <thead>
      <tr>
        <th>id</th>
        <th>title</th>
        <th>owner</th>
        <th>private</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>1</td>
        <td>mockTitle1</td>
        <td>mockOwner</td>
        <td>true</td>
        <td><button>Show post</button></td>
        <td><button>Delete post</button></td>
      </tr>
      <tr>
        <td>2</td>
        <td>mockTitle2</td>
        <td>mockOwner</td>
        <td>false</td>
        <td><button>Show post</button></td>
        <td><button>Delete post</button></td>
      </tr>
    </tbody>
  </table> <button>Create Post</button>
</div>
`;
